File: Q12/Q12.cpp
#include <iostream>
using namespace std;

int main() 
{
    int n;

    cout << "Enter the number of rows: ";
    cin >> n;

    for (int i = n; i >= 1; i--) 
    { 
        for (int j = 1; j <= n - i; j++) 
        {
            cout << " ";
        }

        for (int j = i; j >= 1; j--) 
        {
            cout << j << " ";
        }

        cout << endl; 
    }

    return 0;
}

File: Q12/Q12.js
const n = parseInt(prompt("Enter the number of rows: "));

for (let i = n; i >= 1; i--) 
{
    let spaces = " ".repeat(n - i);
    
    let numbers = "";
    for (let j = i; j >= 1; j--) 
    {
        numbers += j + " ";
    }

    console.log(spaces + numbers);
}
